sdk/core/azure_core_amqp/src/fe2o3/cbs.rs

```diff
@@ -97,12 +97,10 @@ impl AmqpClaimsBasedSecurityApis for Fe2o3ClaimsBasedSecurity<'_> {
                     .to_offset(time::UtcOffset::UTC)
                     .unix_timestamp()
                     .checked_mul(1_000)
-                    .ok_or_else(|| {
-                        azure_core::Error::message(
-                            azure_core::error::ErrorKind::Amqp,
-                            "Unable to convert time to unix timestamp.",
-                        )
-                    })?,
+                    .ok_or(azure_core::Error::message(
```

Comment:
This eagerly allocates the error. The previous way was better.

Reply:
Oh, that's interesting. I had assumed that it would generate dramatically more code because it had to create the closure which does nothing but return an error and that the `ok_or` variant would ultimately reduce the code size. 

I hadn't considered that the compiler would pre-calculate the error even if the value was None (which basically will never happen). 

Reverting.